storage/src/nodestore/alloc.rs

```diff
@@ -666,12 +706,49 @@ mod test_free_list_iterator {
         area_size_index: AreaIndex,
         offset: u64,
     ) {
-        let area: Area<Node, FreeArea> = Area::Free(FreeArea::new(next_free_block));
-        let stored_area = StoredArea::new(area_size_index, area);
-        let stored_area_bytes = serializer().serialize(&stored_area).unwrap();
+        let mut stored_area_bytes = Vec::with_capacity(16);
+        FreeArea::new(next_free_block).as_bytes(area_size_index, &mut stored_area_bytes);
         storage.write(offset, &stored_area_bytes).unwrap();
     }
 
+    #[test]
+    fn test_free_list_decoding_old_format() {
```

Comment:
#[test_case] crate is really nice here:
```suggestion
    #[test_case(&[0x01, 0x01, 0x01, 0x2a], 1, 42), "old format"])
    #[test_case(&[0x02, 0x01, 0x00], 2, 0), "none"])
    #[test_case(&[0x03, 0xff, 0x2b], 3, 43, "new format"]) 
    fn test_free_list_decoding_old_format(data: &[u8], area: u8, addr: u64) {
```

Then, in the function, I'd consider:
 - Calling from_reader and as_bytes and verifying you get the same byte array
 - verify that the area_size matches
 - verify that linear address matches

Probably some other options are possible too.

Reply:
> Probably some other options are possible too

like what?

> Calling from_reader and as_bytes and verifying you get the same byte array

That won't work for this as we're testing deserializing the old format. won't get the same result. And, the other test in this file already does roundtrip serialization of the free node.

I can switch to using test_case for the test verifying the parsed area_size and linear address though.